### PR TITLE
Feature: Support multiple currency sets

### PR DESCRIPTION
**Pull request by varunsrin**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,28 +12,44 @@ to ISO 4217 standards. The main items exported by the library are `Money` and `C
...
+
+
+let v_bucks = Currency::new("VBX", 2)          // Create a US Dollar with an exponent of 2.
+Money::new(1, v_bucks)                         // One V Buck.
+Money::from_major(1, v_bucks)                  // One V Buck.
+Money::from_minor(100, v_bucks)                // One V Buck
+Money::from_str("2,000.00", v_bucks).unwrap(); // One V Buck
+
+// You call also call the money macro for convenience.
+money!(1, v_bucks)                          // One V Buck
```
Comment (jimpo) [inline README.md:31]: Why is this a macro and not just a function?

File: README.md [hunk 2]
```diff
@@ -12,28 +12,44 @@ to ISO 4217 standards. The main items exported by the library are `Money` and `C
...
+money!("1.00", v_bucks)                     // One V Buck
```
[the remark this answers is not in the document]
Comment (varunsrin) [reply README.md:32]: The macro just calls to_string on the input, and then sends them to a Money::from_string call. It's overkill for integers, which get cast twice, but provides a little syntactic sugar for the string use case.  

See here: https://github.com/varunsrin/rusty_money/blob/master/src/money.rs#L28

File: README.md [hunk 3]
```diff
@@ -12,28 +12,44 @@ to ISO 4217 standards. The main items exported by the library are `Money` and `C
...
+ISO-417 Currencies, along with all their metadata and formatting styles. `Currency` is also provided implementing this 
+trait which can be used to represent non-ISO currencies.
 
 ```rust
-// Money can be initialized in a few ways:
 use rusty_money::{money, Money, Currency};
+
+
+let v_bucks = Currency::new("VBX", 2)          // Create a US Dollar with an exponent of 2.
+Money::new(1, v_bucks)                         // One V Buck.
```
[the remark this answers is not in the document]
Comment (varunsrin) [reply README.md:25]: No difference, one is just aliased onto the other. 

It's intentional that none of these constructors can take a float - I want people to convert floats into strings or Decimal types before passing them into money. Does that make sense?

File: src/currency.rs [hunk 4]
```diff
@@ -5,18 +5,97 @@ use std::collections::HashMap;
...
-    static ref CURRENCIES_BY_NUM_CODE: HashMap<String, Currency> =
-        Currency::generate_currencies_by_num_code();
+    static ref CURRENCIES_BY_ALPHA_CODE: HashMap<String, IsoCurrency> =
+        IsoCurrency::generate_currencies_by_alpha_code();
+    static ref CURRENCIES_BY_NUM_CODE: HashMap<String, IsoCurrency> =
+        IsoCurrency::generate_currencies_by_num_code();
 }
 
-/// A struct which represent an ISO-4127 currency.
+pub trait CurrencyType: PartialEq + Eq + Copy {
```
[the remark this answers is not in the document]
Comment (varunsrin) [reply src/currency.rs:14]: The idea is that there is a base Currency type that just does the math and serialization, and a FormattableCurrency type that has a bunch of additional features for localization.

File: src/currency.rs [hunk 5]
```diff
@@ -5,18 +5,97 @@ use std::collections::HashMap;
 use std::fmt;
 
 lazy_static! {
```
[the remark this answers is not in the document]
Comment (varunsrin) [reply src/currency.rs:7]: Not sure, good call out. i will look into it.

File: README.md [hunk 6]
```diff
@@ -86,16 +106,16 @@ The library also provides two additional types - `Exchange` and `ExchangeRates`
...
 
 // Convert 1000 USD to EUR at a 2:1 exchange rate.
-let rate = ExchangeRate::new(Currency::get(USD), Currency::get(EUR), dec!(0.5)).unwrap();
-rate.convert(money!(1000, "USD")); // 500 EUR
+let rate = ExchangeRate::new(IsoCurrency::get(USD), IsoCurrency::get(EUR), dec!(0.5)).unwrap();
+rate.convert(iso_money!(1000, "USD")); // 500 EUR
 
 // An Exchange can be used to store ExchangeRates for later use
 let mut exchange = Exchange::new();
 exchange.add_or_update_rate(&rate);
```
[the remark this answers is not in the document]
Comment (varunsrin) [reply README.md:119]: 👍

File: README.md [hunk 7]
```diff
@@ -86,16 +106,16 @@ The library also provides two additional types - `Exchange` and `ExchangeRates`
...
 use rust_decimal_macros::*;
 
 // Convert 1000 USD to EUR at a 2:1 exchange rate.
-let rate = ExchangeRate::new(Currency::get(USD), Currency::get(EUR), dec!(0.5)).unwrap();
-rate.convert(money!(1000, "USD")); // 500 EUR
+let rate = ExchangeRate::new(IsoCurrency::get(USD), IsoCurrency::get(EUR), dec!(0.5)).unwrap();
+rate.convert(iso_money!(1000, "USD")); // 500 EUR
 
 // An Exchange can be used to store ExchangeRates for later use
 let mut exchange = Exchange::new();
```
[the remark this answers is not in the document]
Comment (varunsrin) [reply README.md:118]: Good points. It doesn't do anything fancy unless there's an exact match on the rate type. I like this idea, but will ad it in a future release.

File: README.md (hunk 1 above)
Comment (varunsrin) [reply README.md:31]: Honestly, for convenience, but I think I'm going to deprecate it.

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (jimpo) [reply README.md:32]: Yeah, you don't need a macro for that, just a generic function `fn money<V: ToString, C: ToString>(value: V, currency: C) -> Money`. Also, the unwrap in the impl is a bad idea IMO. It should just return a `Result`.

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (jimpo) [reply README.md:25]: Yep, that seems fine. But I still think there's no reason to have `new` if it's an alias for `from_major`.

File: src/currency.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (jimpo) [reply src/currency.rs:14]: I think there's most likely a better way to do it without the `CurrencyType` trait.

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (saebs) [reply README.md:32]: @varunsrin this might seem like a potential solution to the Issue i raised for unique situations like that of Zimbabwe currency.

**Pull request merged**